packages/adaptive/src/client.ts

```diff
@@ -0,0 +1,53 @@
+import type { BucketClient } from '@bucketco/browser-sdk';
+import Cookies from 'js-cookie';
+
+const COOKIE_NAME = 'gitbook-visitor-public-bucket';
+
+/**
+ * Sets a client-side session cookie with the features enabled in the bucket client.
+ */
+export function withBucket(client: BucketClient): () => void {
```

Comment:
If this package has something as specific as a bucket integration, we should probably split into 2 packages:

- `@gitbook/adaptive` to write generic data, exports a function `writeGitBookCookie(prefix, data)`
- `@gitbook/adaptive-bucket` to export the `withBucket`

Reply:
The packages are way too small to be split and I don't want to make that decision right now. What are your thoughts about 
 - `@gitbook/adaptive`: for generic data and functions
- `@gitbook/adaptive/bucket`: exports `withBucket`

Reply:
Sure, It can work, I'm not a big fan of this pattern usually, because it means a project will have to download the bucket client dependency, the launchdarkly one, etc, even if they don't use any (I know it'll not be bundled, but still).

> The packages are way too small and simple to be split and I don't want to make that decision right now

This is the whole point of NPM packages: they can be small